### PR TITLE
Remove PoC-1 back-compat hacks

### PR DESCRIPTION
**Pull request by gavofyork**
Remainder of #227:

- Switch to new sandbox externals in runtime WASM
- Switch back to Blake2 and remove tiny-keccak dep
- Remove zero-padding for storage items

### CODE REVIEW

File: substrate/primitives/src/hasher.rs [hunk 1]
```diff
@@ -21,68 +21,34 @@ use plain_hasher::PlainHasher;
...
-// 		type StdHasher = PlainHasher;
-// 		const LENGTH:usize = 32;
-// 		fn hash(x: &[u8]) -> Self::Out {
-// 			blake2_256(x).into()
-// 		}
-// 	}
-// }
-
-pub mod keccak {
+pub mod blake2 {
```
Comment (pepyakin) [inline substrate/primitives/src/hasher.rs]: The comment above is no longer needed?

File: substrate/keystore/src/lib.rs [hunk 2]
```diff
@@ -95,7 +96,7 @@ impl EncryptedKey {
 			.expect("input lengths of key and iv are both 16; qed");
 
 		// KECCAK(DK[16..31] ++ <ciphertext>), where DK[16..31] - derived_right_bits
-		let mac = crypto::derive_mac(&derived_right_bits, &*ciphertext).keccak256();
+		let mac = blake2_256(&crypto::derive_mac(&derived_right_bits, &*ciphertext));
```
Comment (pepyakin) [inline substrate/keystore/src/lib.rs]: Can you update the comment?

File: substrate/primitives/src/rlp_codec.rs [hunk 3]
```diff
@@ -25,17 +25,17 @@ use patricia_trie::{NibbleSlice, NodeCodec, node::Node, ChildReference};
...
 // use BlakeHasher;
-use KeccakHasher;
+use Blake2Hasher;
 
 /// Concrete implementation of a `NodeCodec` with Rlp encoding, generic over the `Hasher`
 pub struct RlpNodeCodec<H: Hasher> {mark: PhantomData<H>}
 
 /// Convenience type for a Keccak/Rlp flavoured NodeCodec
-pub type RlpCodec = RlpNodeCodec<KeccakHasher>;
+pub type RlpCodec = RlpNodeCodec<Blake2Hasher>;
```
Comment (pepyakin) [inline substrate/primitives/src/rlp_codec.rs]: Can you update the comment and remove the commented lines below?

File: substrate/runtime-io/without_std.rs [hunk 4]
```diff
@@ -78,13 +78,13 @@ pub trait ExternTrieCrypto {
 }
 
 // Ensures we use a Keccak-flavoured Hasher when calling into native
-impl ExternTrieCrypto for KeccakHasher {
+impl ExternTrieCrypto for Blake2Hasher {
```
Comment (pepyakin) [inline substrate/runtime-io/without_std.rs]: Can you update the comment?

File: substrate/runtime/executive/src/lib.rs [hunk 5]
```diff
@@ -290,7 +290,7 @@ mod tests {
 		});
 	}
 
-	fn new_test_ext() -> runtime_io::TestExternalities<KeccakHasher> {
+	fn new_test_ext() -> runtime_io::TestExternalities<Blake2Hasher> {
```
Comment (pepyakin) [inline substrate/runtime/executive/src/lib.rs]: Below, in the `block_import_works`, there are comment that blake state root should be `02532989c613369596025dfcfc821339fc9861987003924913a5a1382f87034a`. I wonder why it isn't changed then? Also, can you update the comment there?

File: substrate/state-machine/src/lib.rs [hunk 6]
```diff
@@ -555,7 +555,7 @@ mod tests {
 			b"dogglesworth".to_vec() => b"catXXX".to_vec(),
 			b"doug".to_vec() => b"notadog".to_vec()
 		];
-		let backend = InMemory::<KeccakHasher, RlpCodec>::from(initial);
+		let backend = InMemory::<Blake2Hasher, RlpCodec>::from(initial);
```
Comment (pepyakin) [inline substrate/state-machine/src/lib.rs]: Again, below is commented `blake` hash. I wonder why this weren't changed.

File: substrate/state-machine/src/testing.rs [hunk 7]
```diff
@@ -104,11 +104,11 @@ impl<H: Hasher> Externalities<H> for TestExternalities<H> where H::Out: Ord + En
 #[cfg(test)]
 mod tests {
 	use super::*;
-	use primitives::{KeccakHasher, H256};
+	use primitives::{Blake2Hasher, H256};
 
 	#[test]
 	fn commit_should_work() {
-		let mut ext = TestExternalities::<KeccakHasher>::new();
+		let mut ext = TestExternalities::<Blake2Hasher>::new();
```
Comment (pepyakin) [inline substrate/state-machine/src/testing.rs]: blake/keccak comment below.

File: substrate/test-runtime/src/system.rs [hunk 8]
```diff
@@ -172,9 +172,9 @@ mod tests {
 	use codec::{Joiner, KeyedVec};
 	use keyring::Keyring;
 	use ::{Header, Digest, Extrinsic, Transfer};
-	use primitives::KeccakHasher;
+	use primitives::Blake2Hasher;
 
-	fn new_test_ext() -> TestExternalities<KeccakHasher> {
+	fn new_test_ext() -> TestExternalities<Blake2Hasher> {
```
Comment (pepyakin) [inline substrate/test-runtime/src/system.rs]: blake/keccak comment below, in `block_import_works`.

File: substrate/executor/src/native_executor.rs [hunk 9]
```diff
@@ -203,15 +203,15 @@ macro_rules! native_executor_instance {
 	};
 	(IMPL $name:ident, $dispatcher:path, $version:path, $code:expr) => {
 		// TODO: this is not so great – I think I should go back to have dispatch take a type param and modify this macro to accept a type param and then pass it in from the test-client instead
-		use primitives::KeccakHasher as _KeccakHasher;
+		use primitives::Blake2Hasher as _Blake2Hasher;
```
Comment (dvdplm) [inline substrate/executor/src/native_executor.rs]: @guanqun's recent [PR](https://github.com/paritytech/substrate/pull/683) originally had a fix for this terrible hack now lost after re-working it; perhaps we can sort that out in this PR too?

Comment (gavofyork) [reply substrate/executor/src/native_executor.rs]: second PR would be good if @guanqun fancies putting it in.

**Pull request merged**